### PR TITLE
Add TS examples for getting-started pages of the Auth category

### PR DESCRIPTION
**Pull request by devnotfound**
#### Description of changes:
Add TS examples for getting-started pages of the Auth category
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [X] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 1]
```diff
@@ -2,6 +2,43 @@
 
 Create a new user in the Amazon Cognito UserPool by passing the new user's email address, password, and other attributes to `Auth.signUp`.
 
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+interface SignUpParams {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:11]: @devnotfound, would you mind changing the name of the interface here to TSSignUpParams to preserve the name "SignUpParams" and ensure there's no duplicate declarations of it?

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 2]
```diff
@@ -39,6 +78,29 @@ The `Auth.signUp` promise returns a data object of type [`ISignUpResult`](https:
...
 
 If user didn't get a confirmation code, you can use `resendSignUp` function to send a new one.
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+interface ResendConfCodeParams {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:88]: Same thing here, can we update the usage of `interface` to be a `type` instead?

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 3]
```diff
@@ -51,10 +113,34 @@ async function resendConfirmationCode() {
...
 ### Confirm sign up
 
 If you enabled multi-factor auth, confirm the sign-up after retrieving a confirmation code from the user.
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+interface ConfirmSignUpParams {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:129]: Same thing here, can we update the usage of `interface` to be a `type` instead?

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 4]
```diff
@@ -67,13 +153,39 @@ async function confirmSignUp() {
...
 error. If set to `True`, the API call will migrate the alias from the previous user to 
 a newly created user.
 
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+interface ConfirmSignUpParams {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:171]: Same thing here, can we update the usage of `interface` to be a `type` instead?

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 5]
```diff
@@ -51,10 +113,34 @@ async function resendConfirmationCode() {
...
+  username: string;
+  code: string;
+}
+
+export async function confirmSignUp(confirmSignUpParams: ConfirmSignUpParams) {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:134]: Can you update this to include passing in `username` and `code` as a destructured object such as below:

`confirmSignUp({username, code}:ConfirmSignUpParams)`

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 6]
```diff
@@ -2,6 +2,43 @@
...
+import { Auth } from 'aws-amplify';
+
+interface SignUpParams {
+  username: string;
+  password: string;
+  email: string;
+  phoneNumber: string;
+}
+
+export async function signUp(signUpParams: SignUpParams) {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:18]: Can you update this to include passing in `username`, `password`, `email`,  and `phoneNumber` as a destructured object such as below:

`signUp({username, password, email, phoneNumber}:SignUpParameters)`

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 7]
```diff
@@ -129,7 +243,29 @@ Auth.signUp({
...
+
+```ts
+import { Auth } from 'aws-amplify';
+
+interface SignInParams {
+  username: string;
+  password: string;
+}
+
+export async function signIn(signInParams: SignInParams) {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:257]: Can you update this to include passing in `username` and `password` as a destructured object such as below:

`signIn({username, password}:signInParams)`

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 8]
```diff
@@ -129,7 +243,29 @@ Auth.signUp({
...
 When signing in with user name and password, you will pass in the username and the password to the `signIn` method of the Auth class.
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+interface SignInParams {
```
Comment (cwomack) [inline src/fragments/lib/auth/js/emailpassword.mdx:252]: Same thing here, can we update the usage of `interface` to be a `type` instead?

**Comment by cwomack** [comment]
@devnotfound, greatly appreciate you creating this PR! Left some comments for some updates and feel free to reply back with any questions.

**Comment by devnotfound** [comment]
Hey @cwomack @abdallahshaban557 
I have addressed all the feedback, please let me know if I have missed anything. 
note: I have renamed all the types in the above examples that had the `Params` suffix to `Parameters`

**Review by cwomack — APPROVED** [review]
LGTM!

**Comment by cwomack** [comment]
Appreciate the updates and quick turnaround @devnotfound!

**Review by israx — APPROVED** [review]
LGTM. Thanks for putting this together!

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by katiegoines — APPROVED** [review]

**Pull request merged**